### ISSUE TITLE
Dev.to API response missing articles

### ISSUE THREAD

**Issue by kay-is** [forem/forem#8780]
**Describe the bug**

I do a `fetch()` to https://dev.to/api/articles?username=fllstck 

It responds with only 2 articles on https://fllstck.dev 

**To Reproduce**

Visiting https://fllstck.dev shows 2 articles

Visiting https://dev.to/api/articles?username=fllstck shows 3 articles

**Expected behavior**

I have 3 published articles under the fllstck user.

**Comment by benhalpern** [comment]
Weird. I see two articles in the dev tools for `fllstck.dev` but three when I visit it on its own.

Either way, I suspect that this is an issue with our cache header logic.

**Comment by citizen428** [comment]
Thanks for the issue! We'll take your request into consideration and follow up if we decide to tackle this issue.

To our amazing contributors: issues labeled `type: bug` are always up for grabs, but for feature requests, please wait until we add a `ready for dev` before starting to work on it.

To claim an issue to work on, please leave a comment. If you've claimed the issue and need help, please ping @thepracticaldev/oss and we will follow up within 3 business days.

For full info on how to contribute, please check out our [contributors guide](https://github.com/thepracticaldev/dev.to/blob/master/CONTRIBUTING.md).

**Comment by rhymes** [comment]
I checked the cache headers with 

```shell
curl -svo /dev/null -H "Fastly-Debug: 1" https://dev.to/api/articles\?username\=fllstck
```

and I see that the `Surrogate-Key` contains 3 articles, the cache is miss and the age is 0:

```shell
< Surrogate-Key: articles articles/357562 articles/352555 articles/348887
< X-Cache: MISS, MISS
< X-Cache-Hits: 0, 0
< Age: 0
```

I then issued `curl https://dev.to/api/articles\?username\=fllstck | jq '. | length'` and got `3` as a response.

I then opened https://dev.to/api/articles/?username=fllstck with the browser and see 3 elements.

It might have been a caching issue yeah, probably the cache reset in the meantime

**Comment by rhymes** [comment]
Closed this as it doesn't seem to be an issue anymore

**Comment by arifszn** [comment]
@rhymes , the issue still occurs randomly. In [here](https://penaxr.github.io)(Open the network tab on the debug bar), it shows empty articles but on postman, it shows 1. Is there any workaround to refresh the cache?

![image](https://user-images.githubusercontent.com/45073703/165258620-a368b773-c405-4a90-9185-6fc50da5b1e3.png)

**Comment by maestromac** [comment]
@arifszn this issue is stale and Rhymes is no longer member of the org. Would you mind starting a new issue?

**Comment by arifszn** [comment]
> @arifszn this issue is stale and Rhymes is no longer member of the org. Would you mind starting a new issue?

@maestromac #17531
